Feature/highlight expansion by clicking on bathing area

jedi-of-the-sea opened the pull request:
## Summary

When clicking on a bathing area in the Badegewässer client, the bathing area expansion is highlighted. When clicking on another bathing area or when closing the gfi, the highlighting of the previously selected area is removed.

## Instructions for local reproduction and review

Start the Badegewässer client and click on a bathing area marker. The orange line displaying the bathing area expansion will be highlighted by displaying a thicker line. The highlighting style is set to default if another bathing area is selected or if the gfi is closed.

Sometimes bathing areas are composed of two or more linestrings. To test this, search for "OSTS;GELINGER BUCHT" for example.

EDIT: I checked the emulation for mobile devices in Firefox and realized that the highlighting was not removed, so I adjusted the code accordingly. It should work now, but you might want to check this on an actual mobile phone. 

EDIT EDIT: User can now change tools in iconMenu without deleting the highlighting when a feature is still selected. There might still be a bug which is due to the `select` getter not deselecting when closing the gfi on mobile devices (= the marker stays red and the highlighting is not removed).

## Pull Request Checklist (for Assignee)

- [x] Changelogs are maintained - **the client has not been published yet so there isn't a new entry.**
- [x] Functionality has been tested in Firefox, Chrome, Safari
- [x] Functionality has been tested on a smartphone
- [x] Functionality has been tested with 200% screen zoom
- [x] Screenreader functionality has been manually tested with NVDA

UI has been tested in the following tools regarding accessibility (only regarding functionality affected in this PR)
  - [x] Chrome Lighthouse
  - [x] Firefox Accessibility

## Relevant tickets, issues, et cetera

warm-coolguy reviewed (changes requested):
🏓 @jedi-of-the-sea

jedi-of-the-sea commented on packages/clients/bgw/src/plugins/Gfi/Content.vue:
```diff
@@ -53,11 +59,77 @@ export default Vue.extend({
+    )
+  },
+  beforeDestroy() {
+    if (!this.selected) this.updateBadestraendeStyles(null, this.defaultStyle)
```
This is my fix of the issue I showed you this morning: if a user clicks on layer chooser without unselecting the bathing area, it keep the highlighting. If they close the gfi before that (and deleselect the bathing area), it gets the default style.

warm-coolguy replied:
I've tried to fix it by fixing the `selected` watcher and was really confused. Upon inspection of its neighbouring `currentProperties` watcher, I noticed that `console.warn` statements do not always appear when expected. And finally I realized why: The component is just not there. It's destroyed at times outside of our local control, after all.

To resolve this, I spontaneouly moved everthing to the store, which is never destroyed: https://github.com/Dataport/polar/pull/250/commits/18396b2fe23fa7a28924a2c99971b03a2eaa78d4

Now I'd like you to review back if everything is as it should be. If you think it's good, 🎩.

jedi-of-the-sea commented on packages/clients/bgw/README.md:
```diff
@@ -15,6 +15,7 @@ Usage as a HTML page is simply offered by hosting the client package with the in
 | fieldName | type | description |
 | - | - | - |
 | infoFields | object[] | Array of key-label pair objects. The property of the feature that should be displayed has to be set as `key`. The description of this property has to be set as `label`. |
+| highlightStyleBadestraende | object with attributes `stroke` and / or `fill`| highlight style for the layer 'Badestraende'. |
```
I realized that as I called it 'highlightStyleBadestraende' it would only be used for the badestraende layer. It's a linestring layer, so 'fill' is not needed here. I deleted it and edited the readme.
15e58c2ab22e55ab010f9412de78c32f773f3183
and adjusted the component 84c07194df40ff02139b26e37a2fd38b4f687469

jedi-of-the-sea commented on packages/clients/bgw/src/plugins/Gfi/Content.vue:
```diff
@@ -53,11 +59,77 @@ export default Vue.extend({
+      }
+      if (this.badestraendeFeatures.length > 0) {
+        this.badestraendeFeatures.forEach((feature) => feature.setStyle(style))
+      }
```
d0ed7d61f05faaa144e6110ff339016cd51f9126

jedi-of-the-sea commented on packages/clients/bgw/src/plugins/Gfi/Content.vue:
```diff
@@ -53,11 +59,77 @@ export default Vue.extend({
+        }) ||
+        new Style({
+          stroke: new Stroke({ color: '#FF4500', width: 10 }),
+        })
```
f395582160ad11643b6b7306a2186b7c67b8e2a3

jedi-of-the-sea commented on packages/clients/bgw/src/plugins/Gfi/Content.vue:
```diff
@@ -53,11 +59,77 @@ export default Vue.extend({
+          return feature.get('BATHINGWAT') === badeStellenId
+        })
+    },
+    updateBadestraendeStyles(id: string | null, style: Style) {
```
I did this but I failed to make a specific commit for it >___<

d0ed7d61f05faaa144e6110ff339016cd51f9126

jedi-of-the-sea commented:
🏓 @warm-coolguy

jedi-of-the-sea commented:
🏓 @warm-coolguy I removed the docs script

warm-coolguy commented:
> 🏓 @warm-coolguy I removed the docs script

https://github.com/Dataport/polar/pull/250/commits/67f2a4ce805df01d2ccc0d48c8a9ebd74ca63815

I found another fragment regarding the docs that I removed.

:shipit: @jedi-of-the-sea

jedi-of-the-sea commented:
> > 🏓 @warm-coolguy I removed the docs script
> 
> [67f2a4c](https://github.com/Dataport/polar/commit/67f2a4ce805df01d2ccc0d48c8a9ebd74ca63815)
> 
> I found another fragment regarding the docs that I removed.
> 
> :shipit: @jedi-of-the-sea

Thx for that, this is why I wanted you to have a final look!
